### PR TITLE
Feature/csv reader

### LINKED ISSUE #7: Port projection model from spreadsheet to app integrate with interactive elements
**Is your feature request related to a problem? Please describe.**
As described in the [PRD](https://docs.google.com/document/d/1vssQ4JB8VyfInFpdw7F_4laQG0WVXYv6u4FwIpIFUoQ/edit?ts=5f58fbbc#heading=h.c29ll7o2tvsa), we need to port the projection model from the included spreadsheet to the app. 

**Describe the solution you'd like**
The logic for the model should be ported from the spreadsheet and isolated into a specific directory so that it is easy for non-technical users to find and review the model code later on. As described in the PRD, it should take in the various interactive element selections -- including state, implementation period, and change in revocations -- as well as the data file, and produce the output necessary for the chart and the outcomes section. This should return a single output object that can be passed into the various display elements, including the chart and the Outcomes section.

This will also involve creating the template data file and documenting its spec. This should almost certainly be a basic CSV file that is derived from Parameters tab in the included spreadsheet.

This should also include integrating the interactive elements with the input to the model, and integrating the output of the model with the chart and the Outcomes section. In short, this task wires together all of the elements into a working prototype.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Wmaileq**
## Description of the change

- Added csv to json fetching and parsing utils and corresponding tests
- Updated dev dependencies

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Part of #7 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/params.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+state,revocationsPopulation,populationFraction,totalPopulation,newOffensePopulation,revocationsAdmissions,admissionsFraction,totalAdmissions,newOffenseAdmissions,newOffenseAvgTimeServedInYears,revocationsTimescale,newOffenseA,revocationA,totalCostPerInmate,marginalCostPerInmate,checkpoint1,savings1,checkpoint2,savings2,checkpoint3,savings3,checkpoint4,savings4
+Alabama,672,0.02,33600,32928,3571,0.3,11903.33333,8332.333333,47.42201064,2.258190983,14.64216936,131.7795242,0.01785714286,0.001,672,12,1680,30,2240,40,3360,60
+Alaska,1099,0.25,4396,3297,4895,0.17,28794.11765,23899.11765,1.655458607,2.694177732,1203.046171,151.4067397,0.07279344859,0.001,87.92,6.4,219.8,16,293.0666667,21.33333333,439.6,32
+Texas,22940,0.16,143375,120435,30460,0.47,64808.51064,34348.51064,42.07518676,9.037426133,68.03002213,280.8690545,0.02262423714,0.001,2867.5,64.875,7168.75,162.1875,9558.333333,216.25,14337.5,324.375
```
Comment (jessex) [inline src/params.csv:4]: Nit: need a newline at the end of this file.

File: src/SupervisionSuccess/utils/convertCSVStringToParams.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import csv from "csvtojson";
+
+/**
+ * Functions that extracts needed fields from csv and transforms to params object
```
Comment (jessex) [inline src/SupervisionSuccess/utils/convertCSVStringToParams.js:4]: Let's reword this slightly: "Function that extracts needed fields from the given CSV string and transforms them into parameters ready to be fed into the revocations model."

File: src/SupervisionSuccess/utils/convertCSVStringToParams.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ *   revocationA: number
+ *   revocationsTimescale: number
+ *   marginalCostPerInmate: number
+ *   savingsMap: {
+ *     checkpoint: number
+ *     savings: number
+ *   }[]
+ * }>>}
+ */
+async function convertCSVStringToParams(string) {
```
Comment (jessex) [inline src/SupervisionSuccess/utils/convertCSVStringToParams.js:17]: Nit: rename to `deriveModelParamsFromCsvString`

File: src/SupervisionSuccess/constants.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const CSV_FETCHING_ERROR = "Error occurred while fetching csv file";
```
Comment (jessex) [inline src/SupervisionSuccess/constants.js]: Is this going to be used in a future PR? It doesn't appear to be used now.

File: src/SupervisionSuccess/__tests__/SupervisionSuccess.test.js [hunk 5]
```diff
@@ -1,100 +1,60 @@
...
-      savings: mockSavings,
-      chartData: mockChartData,
-      prisonPopulationDiff: mockPrisonPopulationDiff,
-      finalPopulation: mockFinalPopulation,
-    });
-  });
-
-  it("should change state", () => {
-    render(<SupervisionSuccess params={params} />);
+  it("should render LoadingScreen while loading", () => {
```
Comment (jessex) [inline src/SupervisionSuccess/__tests__/SupervisionSuccess.test.js:38]: It looks like we have a test case for showing the loading screen and for properly mounting once loaded, but can we also add a test case for showing the _error screen_ when an error occurs while processing the CSV?

**Review by jessex — CHANGES_REQUESTED** [review]
Nice work, @Wmaileq! Just some minor remarks and questions from me. Also, make sure to rebase and resolve conflicts now that we've merged #21.

File: src/SupervisionSuccess/constants.js (hunk 4 above)
Comment (Wmaileq) [reply src/SupervisionSuccess/constants.js]: No, but in last version of PR there is no this const.

File: src/SupervisionSuccess/__tests__/SupervisionSuccess.test.js (hunk 5 above)
Comment (Wmaileq) [reply src/SupervisionSuccess/__tests__/SupervisionSuccess.test.js:38]: There is error screen test below

**Review by jessex — APPROVED** [review]
Well done, @Wmaileq! There is still one outstanding question about the CSV file spec itself on the downstream PR #23, which may require changing the column headers referenced in the code, but I think we can address that in that PR specifically. No need to hold up this one.

**Pull request merged**